modules/ITK/cli/statismo-build-deformation-model.cxx

```diff
@@ -0,0 +1,142 @@
+bool isOptionsConflictPresent(programOptions& opt);
+void buildAndSaveDeformationModel(programOptions opt);
+
+
+
+int main(int argc, char** argv) {
+    programOptions poParameters;
+
+    po::positional_options_description optPositional;
+    optPositional.add("output-file", 1);
+    po::options_description optAllOptions = initializeProgramOptions(poParameters);
+
+
+    po::variables_map vm;
+    try {
+        po::parsed_options parsedOptions = po::command_line_parser(argc, argv).options(optAllOptions).positional(optPositional).run();
+        po::store(parsedOptions, vm);
+        po::notify(vm);
+    } catch (po::error& e) {
+        cerr << "An exception occurred while parsing the Command line:"<<endl;
+        cerr << e.what() << endl;
+        return EXIT_FAILURE;
+    }
+
+    if (poParameters.bDisplayHelp == true) {
+        cout << optAllOptions << endl;
+        return EXIT_SUCCESS;
+    }
+    if (isOptionsConflictPresent(poParameters) == true)	{
+        cerr << "A conflict in the options exists or insufficient options were set." << endl;
+        cout << optAllOptions << endl;
+        return EXIT_FAILURE;
+    }
+
+    try {
+        buildAndSaveDeformationModel(poParameters);
+    } catch (ifstream::failure & e) {
+        cerr << "Could not read the data-list:" << endl;
+        cerr << e.what() << endl;
+        return EXIT_FAILURE;
+    } catch (itk::ExceptionObject & e) {
+        cerr << "Could not build the model:" << endl;
+        cerr << e.what() << endl;
+        return EXIT_FAILURE;
+    }
+
+    return EXIT_SUCCESS;
+}
+
+bool isOptionsConflictPresent(programOptions& opt) {
+    if (opt.strDataListFile == "" || opt.strOutputFileName == "" ) {
+        return true;
+    }
+
+    if (opt.strDataListFile == opt.strOutputFileName) {
+        return false;
+    }
+
+    return false;
+}
+
+void buildAndSaveDeformationModel(programOptions opt) {
+    const unsigned Dimensions = 3;
+    typedef itk::Image< itk::Vector<float, Dimensions>, Dimensions > ImageType;
+    typedef itk::StandardImageRepresenter<itk::Vector<float, 3>, 3> RepresenterType;
+    typedef itk::PCAModelBuilder<ImageType> ModelBuilderType;
+    typedef itk::StatisticalModel<ImageType> StatisticalModelType;
+    typedef std::vector<std::string> StringVectorType;
+    typedef itk::DataManager<ImageType> DataManagerType;
+    typedef itk::ImageFileReader<ImageType> ImageReaderType;
+    typedef list<pair<ImageType::Pointer, string> > ImageList;
+
+    RepresenterType::Pointer representer = RepresenterType::New();
+    DataManagerType::Pointer dataManager = DataManagerType::New();
+
+    ImageList images;
+    StringList filenames = getFileList(opt.strDataListFile);
+    for (StringList::const_iterator it = filenames.begin(); it != filenames.end(); it++) {
+        ImageReaderType::Pointer reader = ImageReaderType::New();
+        reader->SetFileName(it->c_str());
+        reader->Update();
+        images.push_back(make_pair(reader->GetOutput(), *it));
```

Comment:
I suppose I using reader->GetFileName() later on instead of saving a pair would be better here.